swift-extensions/UIViewExtensions.swift

```diff
@@ -16,20 +17,13 @@ extension UIView {
 
             bind(viewModel.hidden, \UIView.isHidden)
 
-            let onTapResetableCancelableManager = CancellableManagerProvider()
-            trikotInternalPublisherCancellableManager.add(cancellable: onTapResetableCancelableManager)
-
             if !(self is UIControl) {
                 observe(viewModel.action) {[weak self] (value: ViewModelAction) in
                     guard let self = self else { return }
-                    let newCancellableManager = onTapResetableCancelableManager.cancelPreviousAndCreate()
-
+                    self.removePreviousRegisteredTapAction()
                     if value != ViewModelAction.Companion().None {
-                        let tapGestureReconizer = UITapGestureRecognizer(target: self, action: #selector(self.trikotOnViewTouchUp))
+                        let tapGestureReconizer = TrikotTapGestureRecognizer(target: self, action: #selector(self.trikotOnViewTouchUp))
                         self.addGestureRecognizer(tapGestureReconizer)
```

Comment:
Are you adding a TapGestureRecognizer for buttons too or is it only if it's a control without touch interactions?

Reply:
UIControls and Buttons uses native selector for actions. Its for views that does not have this.